### ISSUE TITLE
Can't link when holding stack of linking books

### ISSUE THREAD

**Issue by dholmes13** [XCompWiz/Mystcraft-Issues#129]
If you right-click with more than one unlinked linking book then nothing happens. Right-clicking with exactly one works as expected.

**Comment by heldplayer** [comment]
That is intended behaviour.

**Comment by dholmes13** [comment]
Ok. It's counter intuitive though because the mod encourages having more than one link book and splitting the stack like that isn't going to be intuitive for everyone. In fact, I definitely thought the mod was broken for about an hour. I suggest either not allowing the unlinked link books to stack, as I'll probably do for my server with some tweaking mod, or having it split off a single linked book from the stack with the right click.

**Comment by keybounce** [comment]
I agree it isn't intuitive. I'm not sure how Vanilla handles a stack of blank maps.

Consider this question: What should Mystcraft do if your inventory were full?

**Comment by dholmes13** [comment]
Vanilla creates a new item in a separate slot when you right click with a stack of blank maps and, in the case where the inventory is full, the new item is thrown in front of the player. The same is true for magic maps in Twilight Forest and the treasure bags in Thaumcraft. I think this is much more intuitive and would want Mystcraft to do the same.

**Comment by keybounce** [comment]
Hmm, and now that I think about it, that was also the behavior when I tried filling a stack of glass bottles with water.

**Comment by XCompWiz** [comment]
The problem is that throwing the book in front of you is different from tossing a map item. The book is an entity.

**Comment by dholmes13** [comment]
Having something that can be thrown out as either an item or an entity depending on some condition isn't without precedent. The pet bat mod is a good example of that. You could just make it throw as an item in this case and as an entity in the normal use case.

**Comment by XCompWiz** [comment]
No, the item form always becomes an entity in the world. I'm not up for special casing that just for this; I think the inconsistency is a bad idea here. thanks for the suggestion, though. :)

**Comment by dholmes13** [comment]
I definitely understand the need to keep things consistent so I can accept that. However, I think having the book thrown out as an entity in that case would still be a significant improvement.

**Comment by XCompWiz** [comment]
Acknowledged.
I would prefer it do nothing on a full inventory, myself. Maybe some kind of error sound or text could be displayed?
Anyone else want to posit an opinion?

**Comment by dholmes13** [comment]
Some indication of the error to the player would be helpful. My main concern is that players will think the mod is broken and then give up before they can witness its glory.

**Comment by heldplayer** [comment]
Maybe use the "pop" sound with a low pitch to indicate it failed?

**Comment by XCompWiz** [comment]
A deep pitched pop would probably be comical. :Pleas

I'll do that.

**Comment by Dmitchell94** [comment]
I agree that it would be much simpler to be able to take a stack and have it split like buckets being filled do. As for if the inventory is full the deep pitched pop and maybe a chat message stating:

Book not linked: Inventory Full

is a great way to solve the problem. I look forward to seeing this in game

**Comment by heldplayer** [comment]
"xComp's words echoed... You can't do everything with a full inventory."
"xComp's words echoed... Your inventory is full."
"xComp's words echoed... You can't do that right now."
"xComp's words echoed... There's a time and place for everything, but not now."
